q2_metadata/_random.py

```diff
@@ -14,14 +14,24 @@
 def shuffle_groups(metadata: qiime2.CategoricalMetadataColumn,
                    n_columns: int = 3,
                    column_name_prefix: str = 'shuffled.grouping.',
-                   column_value_prefix: str = 'fake.group.') -> pd.DataFrame:
+                   # Suggest renaming this variable to sample_value_prefix for
+                   # clarity
+                   column_value_prefix: str = 'fake.group.',
+                   encode_sample_size: bool = False
+                   ) -> pd.DataFrame:
 
     input_column_name = metadata.name
     df = metadata.to_dataframe()
+    group_sample_size = df[input_column_name].value_counts()
 
     value_mapping = {}
     for i, value in enumerate(df[input_column_name].unique()):
-        value_mapping[value] = '%s%d' % (column_value_prefix, i)
+        if encode_sample_size is True:
```

Comment:
Nice work!

Reply:
Since `encode_sample_size` is boolean, you can change this as follows:

```suggestion
        if encode_sample_size:
```

There's no functional difference here, but it's a little more concise.